### PR TITLE
Genome gtf

### LINKED ISSUE #4188: Homo sapiens Genome.gtf test-dataset is invalid
### Have you checked the docs?

- [X] [nf-core website: troubleshooting](https://nf-co.re/usage/troubleshooting)
- [X] [nf-core modules documentation](https://nf-co.re/docs/contributing/modules)

### Description of the bug

The genome.gtf file is not sorted and contains features whose parent feature is absent. Surprisingly, this has not caused issues until now. 

Issue noticed when writing a module for RiboCode (#4019)

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by JackCurragh**
Closes #4188  

Updates to the tests that use the`genome.gtf` test file. 

I sorted the `genome.gtf` using:
```
sed 's/gene/AAA/; s/transcript/BBB/; s/exon/CCC/' genome.gtf | sort -k1,1 -k4,4n -k3,3n | sed 's/AAA/gene/; s/BBB/transcript/; s/CCC/exon/' > genome.gtf.1 
```
and re-ran the tests locally with singularity. 

## PR checklist
- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [X] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by muffato** [comment]
Thanks for opening the pull request, @JackCurragh 
You introduced several `# TODO nf-core: file md5sum was variable, please replace this text with a string found in the file instead` lines in the `test.yml` files. Can you restore the original `contains` rule where there was on, and add one otherwise ?

**Comment by JackCurragh** [comment]
Sorry @muffato I didn't see your reply til now. The original test.yml files had no `-contains` and just had `-path` so I did the same in the latest version

File: tests/modules/nf-core/cellranger/count/test.yml [hunk 1]
```diff
@@ -65,6 +65,4 @@
     - path: output/cellranger/test_10x/outs/raw_feature_bc_matrix/matrix.mtx.gz
       md5sum: 5f0a9e0d5358875030b49eb09afa729c
     - path: output/cellranger/test_10x/outs/web_summary.html
-      contains:
-        - "test_10x"
```
Comment (muffato) [inline tests/modules/nf-core/cellranger/count/test.yml:69]: Is this test not met any more ?

File: tests/modules/nf-core/cellranger/mkref/test.yml [hunk 2]
```diff
@@ -33,7 +33,6 @@
     - path: output/cellranger/homo_sapiens_chr22_reference/star/geneInfo.tab
       md5sum: 8b608537307443ffaee4927d2b428805
     - path: output/cellranger/homo_sapiens_chr22_reference/star/genomeParameters.txt
-      contains: ["genomeGenerate"]
```
Comment (muffato) [inline tests/modules/nf-core/cellranger/mkref/test.yml:36]: Same question here

File: tests/modules/nf-core/cellranger/multi/test.yml [hunk 3]
```diff
@@ -49,8 +45,6 @@
     - path: output/cellranger/homo_sapiens_chr22_reference/star/transcriptInfo.tab
       md5sum: 0c3a5adb49d15e5feff81db8e29f2e36
     - path: output/cellranger/test_10k_pbmc/outs/config.csv
-      contains:
-        - "[libraries]"
```
Comment (muffato) [inline tests/modules/nf-core/cellranger/multi/test.yml:53]: Same question here

File: tests/modules/nf-core/cellranger/multi/test.yml [hunk 4]
```diff
@@ -153,8 +154,6 @@
     - path: output/cellranger/homo_sapiens_chr22_reference/star/geneInfo.tab
       md5sum: 8b608537307443ffaee4927d2b428805
     - path: output/cellranger/homo_sapiens_chr22_reference/star/genomeParameters.txt
-      contains:
-        - "2.7.1a"
```
Comment (muffato) [inline tests/modules/nf-core/cellranger/multi/test.yml:157]: Same question here

File: tests/modules/nf-core/cellranger/multi/test.yml [hunk 5]
```diff
@@ -305,21 +299,28 @@
...
     - path: output/cellranger/test_5k_cmvpos_tcells/outs/per_sample_outs/test_5k_cmvpos_tcells/vdj_t/filtered_contig.fasta
+      md5sum: d41d8cd98f00b204e9800998ecf8427e
     - path: output/cellranger/test_5k_cmvpos_tcells/outs/per_sample_outs/test_5k_cmvpos_tcells/vdj_t/filtered_contig.fastq
+      md5sum: d41d8cd98f00b204e9800998ecf8427e
     - path: output/cellranger/test_5k_cmvpos_tcells/outs/per_sample_outs/test_5k_cmvpos_tcells/vdj_t/filtered_contig_annotations.csv
+      md5sum: d41d8cd98f00b204e9800998ecf8427e
     - path: output/cellranger/test_5k_cmvpos_tcells/outs/per_sample_outs/test_5k_cmvpos_tcells/vdj_t/vdj_contig_info.pb
     - path: output/cellranger/test_5k_cmvpos_tcells/outs/per_sample_outs/test_5k_cmvpos_tcells/web_summary.html
-      contains:
-        - "test_5k_cmvpos_tcells"
```
Comment (muffato) [inline tests/modules/nf-core/cellranger/multi/test.yml:321]: Same question here

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 6]
```diff
@@ -14,52 +15,39 @@
...
     - path: output/simpleaf/human_genome/index/pre_indexing.log
-      contains:
-        - "done building index"
     - path: output/simpleaf/human_genome/index/rank.bin
       md5sum: 3b5c31ce5d3423b15687c59220d8ef3f
     - path: output/simpleaf/human_genome/index/refAccumLengths.bin
       md5sum: ba84af8bcdfe540fb66a59d623f7dfde
     - path: output/simpleaf/human_genome/index/ref_indexing.log
-      contains:
-        - "ntHll estimated 54920 distinct k-mers, setting filter size to 2^20"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:26]: Same question here

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 7]
```diff
@@ -14,52 +15,39 @@
...
     - path: output/simpleaf/human_genome/index/refseq.bin
       md5sum: 64dc6c4822b648c2f36c7934ea9eeeed
     - path: output/simpleaf/human_genome/index/seq.bin
     - path: output/simpleaf/human_genome/index/simpleaf_index.json
-      contains:
-        - "salmon index -k 31"
+      md5sum: b32c91f7706e7d2424f142c11e9da8b2
     - path: output/simpleaf/human_genome/index/t2g_3col.tsv
-      contains:
-        - "ENSG00000100181"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:37]: Same question here

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 8]
```diff
@@ -14,52 +15,39 @@
...
-      contains:
-        - "ntHll estimated 54920 distinct k-mers, setting filter size to 2^20"
     - path: output/simpleaf/human_genome/index/reflengths.bin
       md5sum: 83d53fa7048affe5dfba4ba38f84fd5c
     - path: output/simpleaf/human_genome/index/refseq.bin
       md5sum: 64dc6c4822b648c2f36c7934ea9eeeed
     - path: output/simpleaf/human_genome/index/seq.bin
     - path: output/simpleaf/human_genome/index/simpleaf_index.json
-      contains:
-        - "salmon index -k 31"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:34]: The MD5 doesn't match on Conda. Stick to `contains`

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 9]
```diff
@@ -14,52 +15,39 @@
...
+      md5sum: b32c91f7706e7d2424f142c11e9da8b2
     - path: output/simpleaf/human_genome/index/t2g_3col.tsv
-      contains:
-        - "ENSG00000100181"
     - path: output/simpleaf/human_genome/index/versionInfo.json
       md5sum: f06db51ed32dec928fe518075deb4f2e
     - path: output/simpleaf/human_genome/index_info.json
-      contains:
-        - "bin/alevin-fry"
+      md5sum: 94c97af0ba7e7081366af638e541ef78
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:36]: The MD5 doesn't match on Conda. Stick to `contains`

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 10]
```diff
@@ -14,52 +15,39 @@
...
     - path: output/simpleaf/human_genome/ref/gene_id_to_name.tsv
-      contains:
-        - "ENSG00000287285\tAP000547.4"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:45]: Same question here

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 11]
```diff
@@ -14,52 +15,39 @@
...
     - path: output/simpleaf/human_genome/ref/roers_make-ref.json
       md5sum: 392af1580ee1cb8689353e66b2438ea1
     - path: output/simpleaf/human_genome/ref/roers_ref.fa
       md5sum: b75f577627e23bd7708e2c0276a5299b
     - path: output/simpleaf/human_genome/ref/t2g_3col.tsv
-      contains:
-        - "ENST00000400593"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:52]: Same question here

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 12]
```diff
@@ -14,52 +15,39 @@
...
     - path: output/simpleaf/human_genome/simpleaf_index_log.json
-      contains:
-        - "salmon index -k 31"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:55]: Same question here

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 13]
```diff
@@ -69,35 +57,26 @@
...
     - path: output/simpleaf/human_transcriptome/index/info.json
-      contains:
-        - "human_transcriptome/index"
+      md5sum: 54517a077351434460cbf00c77bec1a0
     - path: output/simpleaf/human_transcriptome/index/mphf.bin
       md5sum: 48234131012798a528048d48881c1ce2
     - path: output/simpleaf/human_transcriptome/index/pos.bin
     - path: output/simpleaf/human_transcriptome/index/pre_indexing.log
-      contains:
-        - "done building index"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:79]: Same question here

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 14]
```diff
@@ -69,35 +57,26 @@
...
     - path: output/simpleaf/human_transcriptome/index/rank.bin
       md5sum: f6e3fc0a3faf4a6e52d31524ddab4c81
     - path: output/simpleaf/human_transcriptome/index/refAccumLengths.bin
       md5sum: d6698204613dfa65fd5d9e426e696745
     - path: output/simpleaf/human_transcriptome/index/ref_indexing.log
-      contains:
-        - "ntHll estimated 50067 distinct k-mers, setting filter size to 2^20"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:86]: Same question here

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 15]
```diff
@@ -69,35 +57,26 @@
...
+      md5sum: c3cabbefec63f2f4bc36a7f9aa1fdd95
     - path: output/simpleaf/human_transcriptome/index/versionInfo.json
       md5sum: f06db51ed32dec928fe518075deb4f2e
     - path: output/simpleaf/human_transcriptome/index_info.json
-      contains:
-        - "bin/alevin-fry"
+      md5sum: 4e7eafb2abad1658d4c5dd6bfc3c0ea7
     - path: output/simpleaf/human_transcriptome/simpleaf_index_log.json
-      contains:
-        - "salmon index -k 31 "
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:102]: Same question here

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 16]
```diff
@@ -69,35 +57,26 @@
     - path: output/simpleaf/human_transcriptome/index/duplicate_clusters.tsv
       md5sum: c96ca031de4888558eec24fd13bd1c9b
     - path: output/simpleaf/human_transcriptome/index/info.json
-      contains:
-        - "human_transcriptome/index"
+      md5sum: 54517a077351434460cbf00c77bec1a0
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:60]: The MD5 doesn't match on Conda. Stick to `contains`

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 17]
```diff
@@ -69,35 +57,26 @@
...
-        - "ntHll estimated 50067 distinct k-mers, setting filter size to 2^20"
     - path: output/simpleaf/human_transcriptome/index/reflengths.bin
       md5sum: 75e3592b2ea434eeadb0ca19d684c206
     - path: output/simpleaf/human_transcriptome/index/refseq.bin
       md5sum: 3d2a23204c63c73bf324ade407a86849
     - path: output/simpleaf/human_transcriptome/index/seq.bin
     - path: output/simpleaf/human_transcriptome/index/simpleaf_index.json
-      contains:
-        - "salmon index -k 31"
+      md5sum: c3cabbefec63f2f4bc36a7f9aa1fdd95
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:76]: The MD5 doesn't match on Conda. Stick to `contains`

File: tests/modules/nf-core/simpleaf/index/test.yml [hunk 18]
```diff
@@ -69,35 +57,26 @@
...
     - path: output/simpleaf/human_transcriptome/index/versionInfo.json
       md5sum: f06db51ed32dec928fe518075deb4f2e
     - path: output/simpleaf/human_transcriptome/index_info.json
-      contains:
-        - "bin/alevin-fry"
+      md5sum: 4e7eafb2abad1658d4c5dd6bfc3c0ea7
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/index/test.yml:80]: The MD5 doesn't match on Conda. Stick to `contains`

File: tests/modules/nf-core/simpleaf/quant/test.yml [hunk 19]
```diff
@@ -1,53 +1,74 @@
...
     - path: output/simpleaf/test_10x/af_map/alevin/alevin.log
-      contains:
-        - "parsing read library format"
     - path: output/simpleaf/test_10x/af_map/aux_info/meta_info.json
-      contains:
-        - "eeb564b47803fd3415a7ef3cca89695fcc0f52dd30eebc381846ff92d7f2c077"
     - path: output/simpleaf/test_10x/af_map/cmd_info.json
-      contains:
-        - "test_10x/af_map"
+      md5sum: b05749e6d1b7d9ba2c1b5f237f174681
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/quant/test.yml:47]: The MD5 doesn't match on Conda. Stick to `contains`

File: tests/modules/nf-core/simpleaf/quant/test.yml [hunk 20]
```diff
@@ -1,53 +1,74 @@
...
-      contains:
-        - "ENSG00000283633"
     - path: output/simpleaf/test_10x/af_quant/alevin/quants_mat_rows.txt
       md5sum: 78e92f0584cc4132374ea7f8fcc1bf1f
     - path: output/simpleaf/test_10x/af_quant/all_freq.bin
       md5sum: ff6a60def164baabaecc05e10b4ac397
     - path: output/simpleaf/test_10x/af_quant/collate.json
-      contains:
-        - "alevin-fry collate"
+      md5sum: 9304d2b2a964c1247c7d18ab18f0b352
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/quant/test.yml:60]: The MD5 doesn't match on Conda. Stick to `contains`

File: tests/modules/nf-core/simpleaf/quant/test.yml [hunk 21]
```diff
@@ -1,53 +1,74 @@
...
     - path: output/simpleaf/test_10x/af_quant/featureDump.txt
       md5sum: 8a337c9fa0660d108dcc0955fcd03b52
     - path: output/simpleaf/test_10x/af_quant/generate_permit_list.json
-      contains:
-        - "alevin-fry generate-permit-list"
+      md5sum: 777485ab9d76ce3cb785ccd648513c5b
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/quant/test.yml:64]: The MD5 doesn't match on Conda. Stick to `contains`

File: tests/modules/nf-core/simpleaf/quant/test.yml [hunk 22]
```diff
@@ -1,53 +1,74 @@
...
     - path: output/simpleaf/test_10x/af_quant/map.collated.rad
       md5sum: bbad29bd96cb245d1893361be81b0ca3
     - path: output/simpleaf/test_10x/af_quant/permit_freq.bin
       md5sum: bfddd006392e272c24849861597c34b4
     - path: output/simpleaf/test_10x/af_quant/permit_map.bin
     - path: output/simpleaf/test_10x/af_quant/quant.json
-      contains:
-        - "alevin-fry quant -i test_10x/af_quant"
+      md5sum: 1af00add44f8e76451f11b62282d5b43
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/quant/test.yml:71]: The MD5 doesn't match on Conda. Stick to `contains`

File: tests/modules/nf-core/simpleaf/quant/test.yml [hunk 23]
```diff
@@ -1,53 +1,74 @@
...
+    - path: output/simpleaf/human/ref/gene_id_to_name.tsv
+    - path: output/simpleaf/human/ref/roers_make-ref.json
+      md5sum: 2b2c8bc03ff255fa2a6b52d9592ee64a
+    - path: output/simpleaf/human/ref/roers_ref.fa
+      md5sum: b75f577627e23bd7708e2c0276a5299b
+    - path: output/simpleaf/human/ref/t2g_3col.tsv
+    - path: output/simpleaf/human/simpleaf_index_log.json
     - path: output/simpleaf/test_10x/af_map/alevin/alevin.log
-      contains:
-        - "parsing read library format"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/quant/test.yml:9]: Same question here

File: tests/modules/nf-core/simpleaf/quant/test.yml [hunk 24]
```diff
@@ -1,53 +1,74 @@
...
     - path: output/simpleaf/test_10x/af_map/aux_info/meta_info.json
-      contains:
-        - "eeb564b47803fd3415a7ef3cca89695fcc0f52dd30eebc381846ff92d7f2c077"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/quant/test.yml:12]: Same question here

File: tests/modules/nf-core/simpleaf/quant/test.yml [hunk 25]
```diff
@@ -1,53 +1,74 @@
...
     - path: output/simpleaf/test_10x/af_map/cmd_info.json
-      contains:
-        - "test_10x/af_map"
+      md5sum: b05749e6d1b7d9ba2c1b5f237f174681
     - path: output/simpleaf/test_10x/af_map/logs/salmon_quant.log
-      contains:
-        - "Number uniquely mapped : 11"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/quant/test.yml:18]: Same question here

File: tests/modules/nf-core/simpleaf/quant/test.yml [hunk 26]
```diff
@@ -1,53 +1,74 @@
...
     - path: output/simpleaf/test_10x/af_map/map.rad
       md5sum: 93ea1875bc36d769ba060d8b38f40aa1
     - path: output/simpleaf/test_10x/af_map/unmapped_bc_count.bin
       md5sum: 7d0f401573b121914df1ef036405187c
     - path: output/simpleaf/test_10x/af_quant/alevin/quants_mat.mtx
-      contains:
-        - "7 21 7"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/quant/test.yml:25]: Same question here

File: tests/modules/nf-core/simpleaf/quant/test.yml [hunk 27]
```diff
@@ -1,53 +1,74 @@
...
     - path: output/simpleaf/test_10x/af_quant/alevin/quants_mat_cols.txt
-      contains:
-        - "ENSG00000283633"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/quant/test.yml:28]: Same question here

File: tests/modules/nf-core/simpleaf/quant/test.yml [hunk 28]
```diff
@@ -1,53 +1,74 @@
...
       md5sum: bfddd006392e272c24849861597c34b4
     - path: output/simpleaf/test_10x/af_quant/permit_map.bin
     - path: output/simpleaf/test_10x/af_quant/quant.json
-      contains:
-        - "alevin-fry quant -i test_10x/af_quant"
+      md5sum: 1af00add44f8e76451f11b62282d5b43
     - path: output/simpleaf/test_10x/af_quant/unmapped_bc_count_collated.bin
     - path: output/simpleaf/test_10x/simpleaf_quant_log.json
-      contains:
-        - "subsampled_5k_human_antiCMV_T_TBNK_connect_GEX_1_S1_L001_R2_001.fastq.gz"
```
Comment (muffato) [inline tests/modules/nf-core/simpleaf/quant/test.yml:52]: Same question here

File: tests/modules/nf-core/spaceranger/count/test.yml [hunk 29]
```diff
@@ -35,15 +35,14 @@
...
+      md5sum: 0527d529622259b09bd46e7f4b49a135
     - path: output/spaceranger/CytAssist_11mm_FFPE_Human_Glioblastoma_2/outs/spatial/tissue_hires_image.png
       md5sum: 1c0f1e94522a886c19f56a629227e097
     - path: output/spaceranger/CytAssist_11mm_FFPE_Human_Glioblastoma_2/outs/spatial/tissue_lowres_image.png
       md5sum: 8c1fcb378f7f886301f49ffc4f84360a
     - path: output/spaceranger/CytAssist_11mm_FFPE_Human_Glioblastoma_2/outs/spatial/tissue_positions.csv
       md5sum: 1b2df34f9e762e9e64aa226226b96c4b
     - path: output/spaceranger/CytAssist_11mm_FFPE_Human_Glioblastoma_2/outs/web_summary.html
-      contains:
-        - "CytAssist_11mm_FFPE_Human_Glioblastoma_2"
```
Comment (muffato) [inline tests/modules/nf-core/spaceranger/count/test.yml:46]: Same question here

File: tests/modules/nf-core/spaceranger/count/test.yml [hunk 30]
```diff
@@ -70,15 +69,14 @@
...
+      md5sum: aad986f7069aa444f71dff8bc6d53712
     - path: output/spaceranger/Visium_FFPE_Human_Ovarian_Cancer/outs/spatial/tissue_hires_image.png
       md5sum: d91f8f176ae35ab824ede87117ac0889
     - path: output/spaceranger/Visium_FFPE_Human_Ovarian_Cancer/outs/spatial/tissue_lowres_image.png
       md5sum: 475a04208d193191c84d7a3b5d4eb287
     - path: output/spaceranger/Visium_FFPE_Human_Ovarian_Cancer/outs/spatial/tissue_positions.csv
       md5sum: 37d288d0e29e8572ea4c5bef292de4b6
     - path: output/spaceranger/Visium_FFPE_Human_Ovarian_Cancer/outs/web_summary.html
-      contains:
-        - "Visium_FFPE_Human_Ovarian_Cancer"
```
Comment (muffato) [inline tests/modules/nf-core/spaceranger/count/test.yml:81]: Same question here

File: tests/modules/nf-core/spaceranger/count/test.yml [hunk 31]
```diff
@@ -110,8 +108,6 @@
     - path: output/spaceranger/homo_sapiens_chr22_reference/star/geneInfo.tab
       md5sum: 8b608537307443ffaee4927d2b428805
     - path: output/spaceranger/homo_sapiens_chr22_reference/star/genomeParameters.txt
-      contains:
-        - "2.7.1a"
```
Comment (muffato) [inline tests/modules/nf-core/spaceranger/count/test.yml:114]: Same question here

File: tests/modules/nf-core/spaceranger/mkref/test.yml [hunk 32]
```diff
@@ -33,7 +33,6 @@
     - path: output/spaceranger/homo_sapiens_chr22_reference/star/geneInfo.tab
       md5sum: 8b608537307443ffaee4927d2b428805
     - path: output/spaceranger/homo_sapiens_chr22_reference/star/genomeParameters.txt
-      contains: ["genomeGenerate"]
```
Comment (muffato) [inline tests/modules/nf-core/spaceranger/mkref/test.yml:36]: Same question here

File: tests/modules/nf-core/star/genomegenerate/test.yml [hunk 33]
```diff
@@ -35,26 +35,4 @@
...
-    - path: output/star/star/chrNameLength.txt
-    - path: output/star/star/chrStart.txt
-    - path: output/star/star/exonGeTrInfo.tab
-    - path: output/star/star/exonInfo.tab
-    - path: output/star/star/geneInfo.tab
-    - path: output/star/star/genomeParameters.txt
-    - path: output/star/star/sjdbInfo.txt
-    - path: output/star/star/sjdbList.fromGTF.out.tab
-    - path: output/star/star/sjdbList.out.tab
-    - path: output/star/star/transcriptInfo.tab
```
Comment (muffato) [inline tests/modules/nf-core/star/genomegenerate/test.yml:60]: Can you restore this `stub` test ?

File: tests/modules/nf-core/stringtie/merge/test.yml [hunk 34]
```diff
@@ -41,29 +43,9 @@
...
-- name: stringtie merge test_stringtie_reverse_merge stub
-  command: nextflow run ./tests/modules/nf-core/stringtie/merge -entry test_stringtie_reverse_merge -c ./tests/config/nextflow.config -c ./tests/modules/nf-core/stringtie/merge/nextflow.config -stub-run
-  tags:
-    - stringtie
-    - stringtie/merge
-  files:
-    - path: output/stringtie/stringtie.merged.gtf
-    - path: output/stringtie/test.coverage.gtf
-    - path: output/stringtie/test.gene.abundance.txt
-    - path: output/stringtie/test.transcripts.gtf
```
Comment (muffato) [inline tests/modules/nf-core/stringtie/merge/test.yml:69]: Can you restore this `stub` test ?

File: tests/modules/nf-core/universc/test.yml [hunk 35]
```diff
@@ -34,7 +35,6 @@
     - path: output/cellranger/homo_sapiens_chr22_reference/star/geneInfo.tab
       md5sum: a20c70b081f5d83649c48ebbd951cb77
     - path: output/cellranger/homo_sapiens_chr22_reference/star/genomeParameters.txt
-      contains: ["genomeGenerate"]
```
Comment (muffato) [inline tests/modules/nf-core/universc/test.yml:37]: Same question here

**Review by muffato — COMMENTED** [review]
Some comments, mostly from looking at the diff:

- The MD5 of empty files (`d41d8cd98f00b204e9800998ecf8427e`) shouldn't be used - it's a convention. Can you remove all the ones you introduced ?
- In several places you removed the existing `contains` rule. I would expect all these checks to still be relevant, so please either restore them, or indicate which ones indeed now break
- Sometimes you replaced a `contains` rule with a `md5sum` check and I can see the test is failing on Conda. Conda is known to behave differently, usually for BAM files, but apparently for some JSON too. Please restore the `contains` rule so that it works across all engines.
- You've removed some `stub` tests. Please restore them.

**Comment by JackCurragh** [comment]
Thank you. I will address those as soon as I can. I did all of this in quite an automated way so I am not sure why these specific things are missing but I can manually reintroduce them

**Comment by JackCurragh** [comment]
@muffato I decided to redo all of this as I only really learned how best to do it as you explained when I went wrong. 

I will merge into this branch in a minute with the changes I made a more systematically.

**Review by muffato — COMMENTED** [review]
Hi @JackCurragh

Big PR but the tests look pretty good.
I think the most important is to fix `arriba/arriba`.

- `rsem/calculateexpression` can be ignored because it's the usual "BAM file is different on Conda"
- `kallistobustools/ref` and `kallistobustools/count` don't work on Conda because they're missing the `requests` package. I think that's outside the scope of this PR and should be reported somewhere else
- I don't know why `msisensor/msi` fails on Conda only. It'd be good if you can find out why (even if that's deferred to a separate issue)

**Review by muffato — APPROVED** [review]

**Comment by muffato** [comment]
- `arriba/arriba` fails on Conda because the BAM files differ. The TSV has got the right rows but in a different order.
- `msisensor/scan` raises a segmentation fault on Conda, making the hole `msisensor/msi` test fails

**Comment by muffato** [comment]
@JackCurragh . I removed the merge from master and your commit that fixes the `environment.yml` files because they were bringing a lot of CI failures (incl. rate limiting). I first want to restore this PR to a clean CI test on its own, and try the merge after.

**Comment by klkeys** [comment]
what is the purpose of this PR?

note that module tests are transitioning from pytest to nf-test: https://nf-co.re/docs/contributing/modules#migrating-from-pytest-to-nf-test

this effort seems unaligned with the future of nf-core modules

**Comment by muffato** [comment]
@klkeys : the GTF file in nf-core/test-datasets has been updated so the module tests (checksums) have to be updated too.

That alone was relatively straightforward, but the PR has flagged (as often the case) many modules whose tests are failing for other reasons. The recent commits are all about fixing those peripheral issues.
Yes, the tests will have to be migrated to nf-test, but the underlying problems would still be there and would still need to be addressed: invalid meta.yml, wrong checksums, missing Conda dependencies, etc.

**Comment by SPPearce** [comment]
I'm going to close this one, as it is now so far behind the current branch that I don;t think it is useful anymore.

**Pull request closed without merging**